src/main/java/com/spotify/annoy/ANNIndex.java

```diff
@@ -8,53 +8,69 @@
 import java.util.*;
 
 /**
- * Read-only Approximate Nearest Neighbor Index which queries databases created by annoy.
+ * Read-only Approximate Nearest Neighbor Index which queries
+ * databases created by annoy.
  */
-public class ANNIndex implements AnnoyIndex {
```

Comment:
Maybe use Guice instead? Not that I know anything about Guice...

On Tue, Mar 8, 2016, 7:51 AM Romain Yon <notifications@github.com> wrote:

> In src/main/java/com/spotify/annoy/ANNIndex.java
> <https://github.com/spotify/annoy-java/pull/2#discussion_r55378485>:
>
> >   */
> > -public class ANNIndex implements AnnoyIndex {
>
> That's a fair point. However we have some mock class in our pipeline code
> that extends this interface and that is injected at test time. I'm not sure
> introducing some new code in the main class strictly for test purposes is
> great.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spotify/annoy-java/pull/2/files#r55378485>.
>